docs/tutorials/tutorial-1.md

```diff
@@ -0,0 +1,766 @@
+
+The answer to the first question is that TSA has several function names of the form `tsa_call_n_m`, where `n` and `m` are numbers. Number `n` indicates the number of results expected in the getter call, and `m` the number of arguments to pass to the getter. For example, `tsa_call_2_3(arg_1, arg_2, arg_3, contract_id, method_id)` means that we expect that `method_id` returns `2` values when calling it with the `3` arguments `arg1, arg2, arg3`. Note that the arguments `contract_id` and `method_id` are not included in the count for number `m` and these two arguments must always be placed at the end of each `tsa_call_n_m` function call.
+
+The answer to the second question is: by an indirect technique that takes advantage on how getter ids are computed by the compiler. We explain next how this indirect technique works. 
+
+The getter id can only be obtained by using an indirect technique because the source code did not declare an explicit id for the `locked` getter, like this:
+
+```tact
+get(3455) fun locked(): Bool {
+    return self.locked;
+}
+```
+where `3455` is the getter id. 
+
+When no id is explicitly declared, the Tact compiler computes a getter id according to the formula:
+
+```tact
+(crc16(<getter_name>) & 0xffff) | 0x10000
+```
+as explained in the [Tact language documentation](https://docs.tact-lang.org/book/functions/#method-ids). 
+
+The FunC compiler also uses the above formula for computing ids for `method_id` declarations. Therefore, by making the empty declaration in the checker contract:
+
+```c
+() locked() method_id { }
+``` 
+we are telling the FunC compiler to compute an id for the name `locked`, which will coincide with the id of the getter in the PeekLevel contract, because they get computed with the same formula.
+
+What remains is to write a function that returns the computed id for the name `locked`. This is achieved by the assembly function:
+
+```c
+int id_locked() asm "locked PUSHINT";
+```
+
+This function works because whenever the name `locked` occurs in assembly code, the compiler replaces the name `locked` for its computed id. Hence, `locked PUSHINT` places the id of `locked` at the top of the TVM stack, so that when function `id_locked` finishes execution, it pops this value and returns it.
+
+> NOTE: If the `locked` getter declared an explicit id, say `get(88503) fun locked(): Bool`, then we would write `tsa_call_1_0(1, 88503)` in the checker contract, without the need to carry out the above indirect technique.
+
+The final part left to explain from the checker contract is:
+
+```c
+throw_if(256, locked_val == 0);
+```
+
+But this is the statement "the value of the `locked` field is `false`" encoded as a failure, as required by the normal behavior of TSA.
+
+In summary, the checker contract is telling TSA to try different messages until it finds one that:
```

Comment:
Minor: it is better to use the wording "telling TSA to find the message that:".

TSA doesn't try random inputs until it finds something that satisfies all conditions, such technique is called fuzzing. Instead, it collects constraints when executing the code with our own interpreter, and only considers inputs that satisfy these constraints.